gqcp/include/Processing/DensityMatrices/SpinResolvedOneDM.hpp

```diff
@@ -89,6 +89,13 @@ class SpinResolvedOneDM {
      */
     const OneDM<Scalar>& beta() const { return this->D_bb; }
 
+    /**
+     *  @return the sum of the norms of the alpha and beta components of the spin resolved one DM
+     */
+    size_t norm() const {
+        return this->alpha().norm() + this->beta().norm();
```

Comment:
You indeed return the sum of the alpha- and beta-norms, but is this really the Frobenius norm of the total, spin-resolved 1-DM?

Reply:
I'll look into correcting this function, hopefully that fixes the issues I've been having!

Reply:
I tried to solve it by creating the generalised density matrix from the alpha and beta components.
```C++
size_t norm() const {

        const auto dim = this->alpha().numberOfOrbitals();
        OneDM<double> generalized_density = OneDM<double>::Zero(dim * 2, dim * 2);

        generalized_density.topLeftCorner(dim, dim) = this->alpha();
        generalized_density.bottomRightCorner(dim, dim) = this->beta();

        return generalized_density.norm();
    }
```
 The result however is exactly the same as with the old function shown above.

Reply:
Since I changed the implementation method, I removed this API for now.

Reply:
Erm... Could it be that you're failing to converge because you're returning `size_t` instead of double?

Reply:
Oh wow, yup, that'll probably be it. Ill change everything accordingly and run the tests. If everything works I'll request your review on this PR one final time.